Poker.BE/Poker.BE.Domain.Tests/Game/TurnTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Poker.BE.Domain.Utility.Exceptions;
+using Poker.BE.Domain.Game;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Poker.BE.Domain.Game.Tests
+{
+    [TestClass()]
+    public class TurnTests
+    {
+        //[TestMethod()]
+        //public void CheckTest()
+        //{
+        //    throw new NotImplementedException();
+        //}
+
+        [TestMethod()]
+        public void CallTest()
+        {
+            //Arrange
+            var player = new Player();
+            var turn = new Turn(player, new Pot(null));
+            Exception expectedException = null;
+
+            //Act
+            try
+            {
+                turn.Call(50);
+            }
+            catch (NotEnoughMoneyException e)
+            {
+                expectedException = e;
+            }
+
+            player.AddMoney(200);
+            turn.Call(50);
+            var res2 = player.Wallet.amountOfMoney != 150;
+
+            //Assert
+            Assert.AreEqual(expectedException.Message, "Player doesn't have enough money!");
+            Assert.IsFalse(res2);
+        }
+
+        [TestMethod()]
+        public void FoldTest()
+        {
+            //Arrange
+            var player = new Player();
+            var turn = new Turn(player, new Pot(null));
+
+            //Act
+            turn.Fold();
+            var res1 = turn.CurrentPlayer.CurrentState == Player.State.ActiveFolded;
+
+            //Assert
+            Assert.IsTrue(res1);
+        }
+
+        [TestMethod()]
+        public void BetTest()
+        {
+            //Arrange
+            var player = new Player();
+            var turn = new Turn(player, new Pot(null));
+            Exception expectedException = null;
+
+            //Act
+            try
+            {
+                turn.Bet(50);
+            }
+            catch (NotEnoughMoneyException e)
+            {
+                expectedException = e;
+            }
+
+            player.AddMoney(200);
+            turn.Bet(50);
+            var res2 = player.Wallet.amountOfMoney != 150;
+
+            //Assert
+            Assert.AreEqual(expectedException.Message, "Player doesn't have enough money!");
```

Comment:
not exactly, I catch a specific type of exception, so a different type of exception won't be caught.